ecolex/legislation.py

```diff
@@ -129,25 +119,40 @@ def harvest_file(upfile):
             key = v['en2'].lower()
             all_languages[key] = v
 
+    resp = requests.get('http://extwprlegs1.fao.org/cgi-bin/xml.exe?database=valid&' \
+                        'search_type=query&table=all&query=T:desext&format_name=@XECOLEX&' \
+                        'lang=xmlf&page_header=@EXMLH&page_footer=@EXMLF&sort_name=@SNAME')
+
+    bs = BeautifulSoup(resp.content.decode(), 'xml')
+    keywords = bs.findAll('dictionary_term')
```

Comment:
You could build a dictionary here `{keyword.Code.string: keyword for keyword in keywords}`, so that you don't iterate over all `keywords` for every `document` and every `field_value` (L 150 - 154).